pkgs/applications/video/gnome-subtitles/default.nix

```diff
@@ -0,0 +1,60 @@
+{ stdenv, fetchFromGitLab, pkg-config, intltool, autoconf, automake, libtool, yelp-tools,
+mono, enchant, gtkspell3, pkgs, gnome3, gtk-doc, itstool, gtk-sharp-3_0, gst_all_1, libxml2 }:
+
+let
+  version = "1.6";
+in stdenv.mkDerivation {
+  pname = "gnome-subtitles";
+  inherit version;
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = "gnome-subtitles";
+    rev = "gnome-subtitles-${version}";
+    sha256 = "1crmmcx32i6ca7dlr3xhnc7vgv9jhlpwh6hxhv2fl1x1zbasf42z";
+  };
+
+  nativeBuildInputs = [ mono intltool itstool pkg-config automake automake libtool yelp-tools ];
+
+  buildInputs = [
+    enchant gtkspell3 gnome3.gnome-common gnome3.gtk
+    gtk-doc libxml2
+    gst_all_1.gstreamer gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good gst_all_1.gst-plugins-bad
+    gst_all_1.gst-libav
+  ];
+
+  propagatedBuildInputs = [ gtk-sharp-3_0 ];
+
+  prePatch = ''
+   patchShebangs autogen.sh
+   substituteInPlace autogen.sh \
+     --replace "which" "${pkgs.which}/bin/which"
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    ./autogen.sh
```

Comment:
Add `autoreconfHook` to `nativeBuildInputs`. This will automatically do the autoconf/automake dance. You could then probably also do with the standard configure, build, and perhaps install phases.